src/main/java/edu/hm/hafner/grading/QualityGate.java

```diff
@@ -0,0 +1,161 @@
+    }
+
+    private final String name;
+    private final String metric;
+    private final double threshold;
+    private final Criticality criticality;
+
+    /**
+     * Creates a new quality gate with the specified parameters.
+     * The comparison operator is automatically determined based on the metric's tendency.
+     *
+     * @param name the name of the quality gate
+     * @param metric the metric to evaluate
+     * @param threshold the threshold value
+     * @param criticality the criticality level
+     */
+    public QualityGate(final String name, final String metric, final double threshold,
+                       final Criticality criticality) {
+        this.name = name;
+        this.metric = metric;
+        this.threshold = threshold;
+        this.criticality = criticality;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getMetric() {
+        return metric;
+    }
+
+    public double getThreshold() {
+        return threshold;
+    }
+
+    public Criticality getCriticality() {
+        return criticality;
+    }
+
+    /**
+     * Evaluates this quality gate against the given actual value.
+     * The comparison direction is automatically determined based on the metric's tendency.
+     *
+     * @param actualValue the actual value to compare against the threshold
+     * @return the evaluation result
+     */
+    public QualityGateEvaluation evaluate(final double actualValue) {
+        boolean passed = isMetricThresholdMet(actualValue);
+        
+        String message = createEvaluationMessage(actualValue, passed);
+        return new QualityGateEvaluation(this, actualValue, passed, message);
+    }
+
+    /**
+     * Determines if the metric value meets the quality gate threshold based on metric tendency.
+     *
+     * @param actualValue the actual metric value
+     * @return true if the value is good (passes the gate), false otherwise
+     */
+    private boolean isMetricThresholdMet(final double actualValue) {
+        try {
+            var metricEnum = Metric.fromName(metric);
+            var tendency = metricEnum.getTendency();
+            
+            if (tendency == Metric.MetricTendency.LARGER_IS_BETTER) {
+                // For coverage, tests, etc. - higher values are better
+                return actualValue >= threshold;
+            } else {
+                // For issues, complexity, etc. - lower values are better  
+                return actualValue <= threshold;
+            }
+        } catch (IllegalArgumentException e) {
+            // Fallback for unknown metrics - assume larger is better
+            return actualValue >= threshold;
+        }
+    }
+
+    /**
+     * Gets the operator symbol based on metric tendency.
+     *
+     * @return ">=" for larger-is-better metrics, "<=" for smaller-is-better metrics
+     */
+    private String getOperatorSymbol() {
+        try {
+            var metricEnum = Metric.fromName(metric);
```

Comment:
I think your default works for warnings as well, but it makes sense to mark this as a future TODO.

```suggestion
            // TODO: currently only Metrics from the coverage-model are supported, but no warnings metrics
            var metricEnum = Metric.fromName(metric);
```